toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -546,48 +763,65 @@ func (im *IsoMaker) buildIsoImageFilePath() string {
 
 // deferIsoMakerCleanUp accepts clean-up tasks to be ran when the entire
 // build process has finished, NOT at the end of the current scope.
-func (im *IsoMaker) deferIsoMakerCleanUp(cleanUpTask func()) {
+func (im *IsoMaker) deferIsoMakerCleanUp(cleanUpTask func() error) {
 	im.isoMakerCleanUpTasks = append(im.isoMakerCleanUpTasks, cleanUpTask)
 }
 
 // isoMakerCleanUp runs all clean-up tasks scheduled through "deferIsoMakerCleanUp".
 // Tasks are ran in reverse order to how they were scheduled.
-func (im *IsoMaker) isoMakerCleanUp() {
+func (im *IsoMaker) isoMakerCleanUp() (err error) {
 	// We defer again to run the tasks in the correct order AND so that a potential
 	// panic from one of these doesn't prevent other ones from running.
 	for _, cleanUpTask := range im.isoMakerCleanUpTasks {
-		defer cleanUpTask()
+		defer func() {
+			cleanupErr := cleanUpTask()
+			if cleanupErr != nil {
+				if err != nil {
+					err = fmt.Errorf("%w\n%w", err, cleanupErr)
+				} else {
+					err = fmt.Errorf("%w", cleanupErr)
```

Comment:
Simplify to: `err = cleanupErr`?

Reply:
I think this may be fine. In case more clean-up tasks fail, we'd lose their errors.

Reply:
If err is nil, it makes sense to have `err = cleanupErr` and not the `err = fmt.Errorf("%w", cleanupErr)`. I think that's what Chris meant. I've updated it to reflect that. Failures from subsequent calls will just accumulate in the err object.